### ISSUE TITLE
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2426)

### ISSUE THREAD

**Issue by monchin** [tlocke/pg8000#163]
Hello, I'm confused about a bug. I'm using `sqlmodel`+`pg8000` to do my postgresql orm, but my project casually raise an Exception. Below is the whole traceback:
```
Exception terminating connection <pg8000.legacy.Connection object at 0x7f75ed19fd90>
Traceback (most recent call last):
  File "/home/polaris/Projects-Code/starion_base/.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 1310, in _checkout
    raise exc.InvalidatePoolError()
sqlalchemy.exc.InvalidatePoolError: ()

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/pg8000/core.py", line 144, in _flush
    sock.flush()
  File "/usr/lib/python3.10/socket.py", line 723, in write
    return self._sock.send(b)                                                                                                                                                                                                                                                     
  File "/usr/lib/python3.10/ssl.py", line 1206, in send
    return self._sslobj.write(data)
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:2426)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py", line 374, in _close_connection
    self._dialect.do_terminate(connection)
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 705, in do_terminate
    self.do_close(dbapi_connection)
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py", line 708, in do_close
    dbapi_connection.close()
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/pg8000/core.py", line 549, in close
    _flush(self._sock)
  File "/home/polaris/Projects-Code/polaris_base/.venv/lib/python3.10/site-packages/pg8000/core.py", line 146, in _flush
    raise InterfaceError("network error") from e
pg8000.exceptions.InterfaceError: network error
```
The most troublesome thing is, the whole traceback doesn't show one single line of my own code, so I even can't know where and why the Exception raises (the program breaks down casually). As it seems that the most direct reason is the `ssl.SSLEOFError (_ssl.c:2426)`, and it is called by `pg8000` `sock.flush()`, I'm wondering if I could get some help or inspiration here to dig out the reason. Thank you!

PS: I found in google that some people got this `SSLEOFError` because of https or proxy, but I'm not using https or proxy neither.

My environment:
- os: ubuntu 22.04
- python: 3.10..12
- pg8000: 1.31.2
- sqlmodel: 0.0.22
- sqlalchemy: 2.0.32

**Comment by tlocke** [comment]
Hi @monchin, thanks for your bug report. If you connect to the database just using pg8000 on its own do you get the same error?

**Comment by monchin** [comment]
> Hi @monchin, thanks for your bug report. If you connect to the database just using pg8000 on its own do you get the same error?

Thank you for your reply! As the error raises casually and I can't know where the error raises in my code from traceback, debugging is quite hard for me. I'm now trying to add some `inspect.frame` info to figure out the reason, but still need to wait for the appearance of the exception. Seems like it raises on the exit of a `with Session` sentence of sqlalchemy. 
I used pure psycopg2 in the past and didn't meet this error. Recently I shifted to pg8000 for more friendly license, together with sqlmodel, change is big so I'm not quite sure why this exception raised. Anyway I need to figure out the whole processure first, although it is difficult. I'll report if there is any progress.
